go/teams/transactions.go

```diff
@@ -334,7 +375,39 @@ func (tx *AddMemberTx) AddMemberByUsername(ctx context.Context, username string,
 		return err
 	}
 
-	return tx.addMemberByUPKV2(ctx, upak.Current, role)
+	_, err = tx.addMemberByUPKV2(ctx, upak.Current, role)
+	return err
+}
+
+// AddMemberByAssertion adds an assertion to the team.
+// Does not attempt to resolve the assertion.
+// Returns (uv, username) which can both be zero-valued if the assertion is not a keybase user.
+func (tx *AddMemberTx) AddMemberByAssertion(ctx context.Context, assertion string,
+	role keybase1.TeamRole) (username libkb.NormalizedUsername, uv keybase1.UserVersion, invite bool, err error) {
+	team := tx.team
+	g := team.G()
+
+	if libkb.NewNormalizedUsername(assertion).CheckValid() == nil {
+		upak, _, err := g.GetUPAKLoader().LoadV2(libkb.NewLoadUserArg(g).WithNetContext(ctx).
+			WithName(assertion).
+			WithPublicKeyOptional().
+			WithForcePoll(true))
+		if err != nil {
+			return "", uv, false, err
+		}
+		invite, err := tx.addMemberByUPKV2(ctx, upak.Current, role)
+		return libkb.NewNormalizedUsername(upak.Current.Username), upak.Current.ToUserVersion(), invite, err
+	}
+	typ, name, err := team.parseSocial(assertion)
```

Comment:
What do you think about making parseSocial not method of `Team` object? a bit nitpicky and probably outside of the scope of this PR. reasons are that it's not even truly a "team" operation, and also its behavior does not depend on given team. We can also add a 0.5p ticket